src/arch.rs

```diff
@@ -0,0 +1,95 @@
+#[derive(Clone, Debug)]
+pub enum Arch {
+    X86,
+    X64,
+    Arm64,
+    Armv7l,
+    Ppc64le,
+    Ppc64,
+    S390x,
+}
+
+#[cfg(unix)]
+/// Get a sane default architecture for the platform.
+pub fn default_str() -> &'static str {
+    use crate::system_info::{platform_arch, platform_name};
+
+    // TODO: Handle (arch, name, version) when Node v15+ supports darwin-arm64
+    match (platform_name(), platform_arch()) {
+        ("darwin", "arm64") => "x64",
+        (_, arch) => arch,
+    }
+}
+
+#[cfg(windows)]
+/// Get a sane default architecture for the platform.
+pub fn default_str() -> &'static str {
+    return crate::system_info::platform_arch();
+}
+
+pub fn get_default() -> Arch {
+    match from_str(default_str()) {
+        Ok(arch) => arch,
+        Err(e) => panic!("{}", e.details),
+    }
+}
+
+fn from_str(s: &str) -> Result<Arch, ArchError> {
+    match s {
+        "x86" => Ok(Arch::X86),
+        "x64" => Ok(Arch::X64),
+        "arm64" => Ok(Arch::Arm64),
+        "armv7l" => Ok(Arch::Armv7l),
+        "ppc64le" => Ok(Arch::Ppc64le),
+        "ppc64" => Ok(Arch::Ppc64),
+        "s390x" => Ok(Arch::S390x),
+        unknown => Err(ArchError::new(&format!("Unknown Arch: {}", unknown))),
+    }
+}
```

Comment:
This can go into the `FromStr` impl block

Reply:
I had it this way... then needed to parse in two places so extracted... 

But TIL .parse() is magical.  Surprised how well the type inference works!  I feel like Go and OCAML got drunk and and had a baby and called it Rust.

Reply:
53e1a4c